<launch>

	<include
		file="$(find autonomous)/launch/camera.launch"
	/>

	
	<node
		pkg="autonomous"
		name="diagnostic_node"
		type="diagnostic_node.py"
		respawn="true"
	/>


	<node
		pkg="autonomous"
		name="image_preprocessing_node"
		type="image_preprocessing_node.py"
		respawn="true"
	/>

	
	<node
		pkg="autonomous"
		name="image_lanedetection_node"
		type="image_lanedetection_node.py"
		respawn="true"
	/>

	
	<node
		pkg="autonomous"
		name="image_stopsign_node"
		type="image_stopsign_node.py"
		respawn="true"
	/>

	
	<node
		pkg="autonomous"
		name="esc_control_node"
		type="esc_control_node.py"
		respawn="true"
	/>



	<node
		pkg="rosserial_python"
		name="serial_node"
		type="serial_node.py"
		args="/dev/ttyACM0"
		respawn="true"
	/>

	
	<node name="steering_pid" pkg="pid" type="controller" >
		<param name="Kp" value="5.0" />
		<param name="Ki" value="1.0" />
		<param name="Kd" value="0.0" />
		<param name="upper_limit" value="20" />
		<param name="lower_limit" value="-20" />
		<param name="topic_from_controller" value="control_steering" />
		<param name="topic_from_plant" value="abw_steering" />
		<param name="setpoint_topic" value="set_steering" />
	</node>


</launch>
